### ISSUE TITLE
Illegal values in optional_permissions should be accepted

### ISSUE THREAD

**Issue by piroor** [mozilla/addons-linter#1788]
### Describe the problem and steps to reproduce it:

Firefox now accepts illegal values in `optional_permissions` (after the [Bug 1350759: Warn but accept illegal values in optional_permissions](https://bugzilla.mozilla.org/show_bug.cgi?id=1350759)) but the linter still blocks any unknown permission names. Thus I cannot release [my addon including new permission `tabHide` in optional permissions](https://github.com/piroor/treestyletab/blob/157d5087535d64df1490666c5256a5c2890959e0/webextensions/manifest.json#L18) which is [implemented at Firefox 59](https://bugzilla.mozilla.org/show_bug.cgi?id=1423725). 

Steps to reproduce:

 1. Create an addon including `tabHide` in its `optional_permissions`.
 2. Upload it to AMO.

### What happened?

I got 4 errors and couldn't release the version:

```
"/optional_permissions/2" should be equal to one of the allowed values

エラー: Your JSON file could not be parsed.
"/optional_permissions/2" should match some schema in anyOf

エラー: Your JSON file could not be parsed.
"/optional_permissions/2" should match pattern "^(https?|wss?|file|ftp|\*)://(\*|\*\.[^*/]+|[^*/]+)/.*$"

エラー: Your JSON file could not be parsed.
"/optional_permissions/2" should match pattern "^file:///.*$"

エラー: Your JSON file could not be parsed.
```

### What did you expect to happen?

A new permission `tabHide` is warned but not reported as an error.

### Anything else we should know?
(Please include screenshots and any relevant files.)

This is the file I uploaded.
[treestyletab-we.xpi.zip](https://github.com/mozilla/addons-linter/files/1663784/treestyletab-we.xpi.zip)

**Comment by muffinresearch** [comment]
Is this still reproducible for optional permissions?
